### ISSUE TITLE
segmentation fault

### ISSUE THREAD

**Issue by alima90** [icebert/pblat#7]
Hi

I'm running pblat for running AUGUSTUS gene prediction
however, segmentation fault error was occurred during running pblat

I used this command
pblat -threads=40 -t=dnax -q=prot -noHead -minIdentity=81 -minScore=15 genome.fa protein.fa output.out

The size of genome.fa is 2.4G and the size of protein.fa is 8.8M
please let me know how to solve the problem
Thanks

**Comment by hmehlan** [comment]
I have a similar problem on Ubuntu 18.04.3 and Debian 9.6 (x86_64) with pblat v. 36x2

pblat -t=dnax -q=prot -threads=2 -minIdentity=81 -minScore=15 genome.fa protein.fa out.psl

finished for -threads=2 (and more) with 
```
  Loaded 560 letters in 1 sequences
  Blatx 1 sequences in database, 2 files in query
  free(): double free detected in tcache 2
  Aborted (core dumped) 
```
and for -threads=1 with 
```
  Loaded 560 letters in 1 sequences
  Blatx 1 sequences in database, 3 files in query
  Segmentation fault (core dumped)

```

with genome.fa:
```
>test
TTCTGTTTCTATTTTGTGGTTACTTTGAGGAGAGTTGGAATTAGGTCTTCTTTGAAGGTCTGGTAGAACT
CTGCATTAAACCCATCTGGTCCTGGGCTTTTTTTTTTTTTTTTTTTTTTTTTTGGGTGGGAGACTATTGA
TGACTGCCTCTATTTCTTTAGGGGAAATGGGACTTTTAGTCCATGAATCTGATCCTGATTTAGCTTTGGT
ACCTGGTATCTGTCTAGGAAGTTGTCCATTTCATCCAGGTTTTCCTGGTTTTTTTTTAGTATAGCCTTTC
ATAGTAAAATCTGATGATGTTTTTGATATCCTCATGTTCTGTTGGTATGTCTCCTTTTTCATTTCTGATT
TTGTTAATTATAGTACAGTCCCTATGCCCTCTAGTTAGTCTGGCTAAGGGTTTATCTATCTTGTTGACTT
TCTCAAAGAACCAGCTACTATTTTGGTTGATTCTTTGAATATTTCTTTTTGTTTCCACTTGGTTGATTTC
AGCTCTGAGTTTGATTATTTCCTGCTGTCTACTCATCTTGGGTGAATTTGCTTCCTTTTGTTCTAGAGCT
```

and protein.fa:
```
>testA
MGAYKYIQELWRKKQSDVMRFLLRVRCWQYRQLSALHRAPRPTWPDKARRLGYKAKQGYVIYRIRVRRGG
RKRPVPKGATYGKPVHHGVNQLKFARSLQSVAEERAGRHCGALRVLNSYWVGEDSTYKFFEVILIDPFHK
AIRRNPDTQWITKPVHKHREMRGLTSAGRKSRGLGKGHKFHHTIGGSRRAAWRRRNTLQLHRYR
>testB
MAHSERANGLQESNQRYGSLQEQLPKAGNQAVGPIEIFRFADNLDIVLMTLGILASMINGATVPLMSLVL
GEISDHFINGCLVQTNKTKYQNCSQSQEKLNEDIIMLTLYYVGIGAAALVLGYVQISFWVITAARQTTRI
RKQFFHSILAQDISWFDGTDICELNTRMNGDISKLCDGIGDKIPLMFQNISGFSIGLVISLIKSWKLSLA
ILSTSPLIMAASALCSRMVISLTSKELDAYSKAGAVAEEALSSIRTVTAFGAQEKEIQRYTQNLKDAKDA
```

The files are artificial just to reproduce the error, it happens with much larger genome and protein files too.
The underlying blat program handle this input correct.

Many thanks

PS:
while pblat on ubuntu displays just the messages above, pblat in a singularity container running Debian 9.6 is more verbose:

```
Loaded 560 letters in 1 sequences
Blatx 1 sequences in database, 2 files in query
*** Error in `pblat': double free or corruption (out): 0x00005605861ad320 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x70bfb)[0x7f4a999d1bfb]
/lib/x86_64-linux-gnu/libc.so.6(+0x76fc6)[0x7f4a999d7fc6]
/lib/x86_64-linux-gnu/libc.so.6(+0x7780e)[0x7f4a999d880e]
pblat(+0x79da)[0x5605846d29da]
pblat(+0x7e82)[0x5605846d2e82]
pblat(+0x85f2)[0x5605846d35f2]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf1)[0x7f4a999812e1]
pblat(+0x674a)[0x5605846d174a]
======= Memory map: ========
5605846cb000-5605847b2000 r-xp 00000000 07:04 21518                      /usr/local/share/pblat/pblat
5605849b2000-5605849b3000 r--p 000e7000 07:04 21518                      /usr/local/share/pblat/pblat
5605849b3000-5605849b6000 rw-p 000e8000 07:04 21518                      /usr/local/share/pblat/pblat
5605849b6000-5605849c9000 rw-p 00000000 00:00 0 
56058618c000-56058cfcb000 rw-p 00000000 00:00 0                          [heap]
7f4a84000000-7f4a84021000 rw-p 00000000 00:00 0 
7f4a84021000-7f4a88000000 ---p 00000000 00:00 0 
7f4a8c000000-7f4a8c021000 rw-p 00000000 00:00 0 
7f4a8c021000-7f4a90000000 ---p 00000000 00:00 0 
7f4a91978000-7f4a91979000 ---p 00000000 00:00 0 
7f4a91979000-7f4a92179000 rw-p 00000000 00:00 0 
7f4a92179000-7f4a9217a000 ---p 00000000 00:00 0 
7f4a9217a000-7f4a9297a000 rw-p 00000000 00:00 0 
7f4a99546000-7f4a9955c000 r-xp 00000000 07:04 1495                       /lib/x86_64-linux-gnu/libgcc_s.so.1
7f4a9955c000-7f4a9975b000 ---p 00016000 07:04 1495                       /lib/x86_64-linux-gnu/libgcc_s.so.1
7f4a9975b000-7f4a9975c000 r--p 00015000 07:04 1495                       /lib/x86_64-linux-gnu/libgcc_s.so.1
7f4a9975c000-7f4a9975d000 rw-p 00016000 07:04 1495                       /lib/x86_64-linux-gnu/libgcc_s.so.1
7f4a9975d000-7f4a99760000 r-xp 00000000 07:04 1485                       /lib/x86_64-linux-gnu/libdl-2.24.so
7f4a99760000-7f4a9995f000 ---p 00003000 07:04 1485                       /lib/x86_64-linux-gnu/libdl-2.24.so
7f4a9995f000-7f4a99960000 r--p 00002000 07:04 1485                       /lib/x86_64-linux-gnu/libdl-2.24.so
7f4a99960000-7f4a99961000 rw-p 00003000 07:04 1485                       /lib/x86_64-linux-gnu/libdl-2.24.so
7f4a99961000-7f4a99af6000 r-xp 00000000 07:04 1471                       /lib/x86_64-linux-gnu/libc-2.24.so
7f4a99af6000-7f4a99cf6000 ---p 00195000 07:04 1471                       /lib/x86_64-linux-gnu/libc-2.24.so
7f4a99cf6000-7f4a99cfa000 r--p 00195000 07:04 1471                       /lib/x86_64-linux-gnu/libc-2.24.so
7f4a99cfa000-7f4a99cfc000 rw-p 00199000 07:04 1471                       /lib/x86_64-linux-gnu/libc-2.24.so
7f4a99cfc000-7f4a99d00000 rw-p 00000000 00:00 0 
7f4a99d00000-7f4a99f6a000 r-xp 00000000 07:04 16919                      /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
7f4a99f6a000-7f4a9a16a000 ---p 0026a000 07:04 16919                      /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
7f4a9a16a000-7f4a9a188000 r--p 0026a000 07:04 16919                      /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
7f4a9a188000-7f4a9a196000 rw-p 00288000 07:04 16919                      /usr/lib/x86_64-linux-gnu/libcrypto.so.1.1
7f4a9a196000-7f4a9a199000 rw-p 00000000 00:00 0 
7f4a9a199000-7f4a9a1fc000 r-xp 00000000 07:04 17417                      /usr/lib/x86_64-linux-gnu/libssl.so.1.1
7f4a9a1fc000-7f4a9a3fb000 ---p 00063000 07:04 17417                      /usr/lib/x86_64-linux-gnu/libssl.so.1.1
7f4a9a3fb000-7f4a9a3ff000 r--p 00062000 07:04 17417                      /usr/lib/x86_64-linux-gnu/libssl.so.1.1
7f4a9a3ff000-7f4a9a405000 rw-p 00066000 07:04 17417                      /usr/lib/x86_64-linux-gnu/libssl.so.1.1
7f4a9a405000-7f4a9a41e000 r-xp 00000000 07:04 1587                       /lib/x86_64-linux-gnu/libz.so.1.2.8
7f4a9a41e000-7f4a9a61d000 ---p 00019000 07:04 1587                       /lib/x86_64-linux-gnu/libz.so.1.2.8
7f4a9a61d000-7f4a9a61e000 r--p 00018000 07:04 1587                       /lib/x86_64-linux-gnu/libz.so.1.2.8
7f4a9a61e000-7f4a9a61f000 rw-p 00019000 07:04 1587                       /lib/x86_64-linux-gnu/libz.so.1.2.8
7f4a9a61f000-7f4a9a637000 r-xp 00000000 07:04 1556                       /lib/x86_64-linux-gnu/libpthread-2.24.so
7f4a9a637000-7f4a9a836000 ---p 00018000 07:04 1556                       /lib/x86_64-linux-gnu/libpthread-2.24.so
7f4a9a836000-7f4a9a837000 r--p 00017000 07:04 1556                       /lib/x86_64-linux-gnu/libpthread-2.24.so
7f4a9a837000-7f4a9a838000 rw-p 00018000 07:04 1556                       /lib/x86_64-linux-gnu/libpthread-2.24.so
7f4a9a838000-7f4a9a83c000 rw-p 00000000 00:00 0 
7f4a9a83c000-7f4a9a93f000 r-xp 00000000 07:04 1514                       /lib/x86_64-linux-gnu/libm-2.24.so
7f4a9a93f000-7f4a9ab3e000 ---p 00103000 07:04 1514                       /lib/x86_64-linux-gnu/libm-2.24.so
7f4a9ab3e000-7f4a9ab3f000 r--p 00102000 07:04 1514                       /lib/x86_64-linux-gnu/libm-2.24.so
7f4a9ab3f000-7f4a9ab40000 rw-p 00103000 07:04 1514                       /lib/x86_64-linux-gnu/libm-2.24.so
7f4a9ab40000-7f4a9ab63000 r-xp 00000000 07:04 1447                       /lib/x86_64-linux-gnu/ld-2.24.so
7f4a9ad52000-7f4a9ad58000 rw-p 00000000 00:00 0 
7f4a9ad62000-7f4a9ad63000 rw-p 00000000 00:00 0 
7f4a9ad63000-7f4a9ad64000 r--p 00023000 07:04 1447                       /lib/x86_64-linux-gnu/ld-2.24.so
7f4a9ad64000-7f4a9ad65000 rw-p 00024000 07:04 1447                       /lib/x86_64-linux-gnu/ld-2.24.so
7f4a9ad65000-7f4a9ad66000 rw-p 00000000 00:00 0 
7fff93761000-7fff93783000 rw-p 00000000 00:00 0                          [stack]
7fff937f0000-7fff937f3000 r--p 00000000 00:00 0                          [vvar]
7fff937f3000-7fff937f5000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
Aborted (core dumped)
```

**Comment by icebert** [comment]
Hi alima90 and hmehlan, currently the pblat doesn't support querying protein sequences against genome sequences in multithreads mode. I'll try to implement this in the future. Currently it only supports querying nucleotide sequences against genome sequences in multithreads.

**Comment by aagatam** [comment]
Hello,

I also encountered segmentation fault when trying to align reads to mouse genome:
 pblat -threads=32 -t=dna -q=rna genome.fa reads.fa output.psl
the size of the genome is 12GB and the reads.fa size is 2.3GB. Do you know what could be the problem?

Thanks

**Comment by icebert** [comment]
I fixed this in the new release (v2.5). Please compile with the latest code or install with bioconda.
